src/ripple/ledger/PaymentSandbox.h

```diff
@@ -188,6 +188,13 @@ class PaymentSandbox final
     apply (PaymentSandbox& to);
     /** @} */
 
+    // Return a map of balance changes on trust lines and amount of XRP destoryed.
+    // The low account is be the first account in the key.
+    // If the two accounts are equal, the map contains the total changes in currency
+    // regargless of issuer. This is useful to get the total change in XRP balances.
+    std::pair<std::map<std::tuple<AccountID, AccountID, Currency>, STAmount>, XRPAmount>
```

Comment:
Not excited about returning a pair here.  To me it makes more sense to have two different methods: `balancesChanges` returns the map and a `dropsDestroyed` method to return the destroyed XRP.  The implementation doesn't suggest that there's an optimization from computing both together (although I could be wrong on that).  "5. Give one entity one cohesive responsibility" -- Sutter and Alexandrescu